File: pinyougou/pinyougou-manage-web/src/main/webapp/js/controller/brandController.js
//定义处理器
app.controller("brandController", function ($scope, $http,$controller, brandService) {

    //继承baseController；参数1：要继承的处理器名称，参数2：将本处理器的scope传递到父控制器
    $controller("baseController", {$scope:$scope});

    $scope.findAll = function () {

        brandService.findAll().success(function (response) {
            //将返回的品牌列表设置到一个变量中
            $scope.list = response;
        }).error(function () {
            alert("加载数据失败！");
        });

    };

    //根据分页查询
    $scope.findPage = function (page, rows) {
        brandService.findPage(page, rows)
            .success(function (pageResult) {
                //更新list
                $scope.list = pageResult.rows;
                //更新分页插件中的总记录数
                $scope.paginationConf.totalItems = pageResult.total;
            });
    };

    //保存
    $scope.save = function () {

        var obj;
        //如果是修改的话；则修改为update
        if($scope.entity.id != null){
            obj = brandService.update($scope.entity);
        } else {
            //新增
            obj = brandService.add($scope.entity);
        }

        obj.success(function (response) {
            if(response.success){
                //操作成功；则刷新列表
                $scope.reloadList();
                //$scope.entity={};
            } else {
                alert(response.message);
            }
        });
    };

    //根据主键查询
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        });
    };

    //删除
    $scope.delete = function () {
        //判断是否选择了
        if($scope.selectedIds.length == 0){
            alert("请先选择要删除的记录");
            return;
        }

        //如果点击了 确定 则返回true；否则false
        if(confirm("确定要删除选中的那些记录吗？")){
            brandService.delete($scope.selectedIds).success(function (response) {
                if(response.success){
                    $scope.reloadList();
                    //将已选择的id清空
                    $scope.selectedIds = [];
                } else {
                    alert(response.message);
                }
            });
        }

    };

    $scope.searchEntity = {};

    //查询
    $scope.search = function (page, rows) {
        brandService.search(page, rows, $scope.searchEntity).success(function (response) {
            $scope.paginationConf.totalItems = response.total;
            $scope.list = response.rows;
        });

    };

});

File: pinyougou/pinyougou-manage-web/src/main/java/com/pinyougou/manage/controller/BrandController.java
package com.pinyougou.manage.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import com.pinyougou.vo.PageResult;
import com.pinyougou.vo.Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RequestMapping("/brand")
/*@Controller*/
@RestController //组合注解 @Controller @ResponseBody
public class BrandController {

    //从注册中心返回一个代理对象
    @Reference
    private BrandService brandService;

    /**
     * 根据分页参数和查询条件查询品牌数据
     * @param brand 查询条件
     * @param page 页号
     * @param rows 页大小
     * @return 分页对象
     */
    @PostMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,
                             @RequestParam(defaultValue = "1")Integer page,
                             @RequestParam(defaultValue = "10")Integer rows){
        return brandService.search(brand, page, rows);
    }


    /**
     * 根据品牌id数组批量删除品牌
     * @param ids 品牌id数组
     * @return 操作结果
     */
    @GetMapping("/delete")
    public Result delete(Long[] ids){

        try {
            brandService.deleteByIds(ids);
            return Result.ok("删除成功");
        } catch (Exception e) {
            e.printStackTrace();
        }

        return Result.fail("删除失败");
    }

    /**
     * 根据品牌id查询品牌数据
     * @param id 品牌id
     * @return 品牌数据
     */
    @GetMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    /**
     * 根据品牌id将品牌更新到数据库中
     * @param brand 品牌
     * @return 操作结果
     */
    @PostMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            //return new Result(true, "新增品牌成功");
            return Result.ok("更新品牌成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("更新品牌失败");
    }

    /**
     * 将品牌保存到数据库中
     * @RequestBody 利用springmMVC的消息转换器将前端传递的json格式字符串
     * 转换为java对象
     * @param brand 品牌
     * @return 操作结果
     */
    @PostMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try {
            brandService.add(brand);
            //return new Result(true, "新增品牌成功");
            return Result.ok("新增品牌成功");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return Result.fail("新增品牌失败");
    }

    /**
     * 根据分页信息查询品牌列表
     * @param page 页号
     * @param rows 页大小
     * @return 品牌列表
     */
    @GetMapping("/findPage")
    public PageResult findPage(@RequestParam(defaultValue = "1") Integer page,
                               @RequestParam(defaultValue = "10") Integer rows){
        return brandService.findPage(page, rows);
    }


    /**
     * 根据分页信息查询品牌列表
     * @param page 页号
     * @param rows 页大小
     * @return 品牌列表
     */
    @GetMapping("/testPage")
    public List<TbBrand> testPage(@RequestParam(defaultValue = "1") Integer page, @RequestParam(defaultValue = "20") Integer rows){
        //return brandService.testPage(page, rows);
        return (List<TbBrand>) brandService.findPage(page, rows).getRows();
    }

    /**
     * 测试查询全部品牌数据
     * @return 品牌列表json字符串
     */
    /*@RequestMapping(value="/findAll", method = RequestMethod.GET)
    @ResponseBody*/
    @GetMapping("/findAll")
    public List<TbBrand> findAll(){
        //return brandService.queryAll();
        return brandService.findAll();
    }

    /**
     * 根据品牌列表；数据结构如：[{id:'1',text:'联想'},{id:'2',text:'华为'}]
     * @return 有格式的品牌列表数据
     */
    @GetMapping("/selectOptionList")
    public List<Map<String, Object>> selectOptionList(){
        return brandService.selectOptionList();
    }
}
